### PR TITLE
Add enum extension

### PR DESCRIPTION
**Pull request by AllyW**

### CODE REVIEW

**Comment by AllyW** [comment]
image

File: src/aaz_dev/cli/controller/az_arg_group_generator.py [hunk 1]
```diff
@@ -172,7 +172,7 @@ def parse_arg_help(help):
     if not help.lines and not help.ref_commands:
         if not help.short:
             raise exceptions.InvalidAPIUsage("Invalid argument help, short summary is miss.")
-        return help.short
+        return help.short.replace("\r", "").replace("\n", "")
```
Comment (kairu-ms) [inline src/aaz_dev/cli/controller/az_arg_group_generator.py]: For short summary, I think it's better keep only the first line. And append the rests into long summary when generate the arg help. Please do not add these logic in the cli generators. It should be fixed when generate the commands.

Comment (kairu-ms) [reply src/aaz_dev/cli/controller/az_arg_group_generator.py]: You can revoke the change and arise another PR.

**Comment by kairu-ms** [comment]
please update the doc as well

**Comment by AllyW** [comment]
image

File: src/web/src/views/workspace/WSEditorCommandArgumentsContent.tsx [hunk 2]
```diff
@@ -2175,6 +2190,8 @@ function decodeArg(response: any): { arg: CMDArg, clsDefineMap: ClsArgDefinition
         idPart: response.idPart,
         prompt: prompt,
         configurationKey: response.configurationKey,
+        supportEnumExtension: response.enum?.support_extension || response.item?.enum?.support_extension || false,
```
Comment (kairu-ms) [inline src/web/src/views/workspace/WSEditorCommandArgumentsContent.tsx]: I think it should be `supportExtension` in response. Please verify it.
```suggestion
        supportEnumExtension: response.enum?.supportExtension || response.item?.enum?.supportExtension || false,
```

**Comment by AllyW** [comment]
image

File: src/aaz_dev/command/model/configuration/_arg.py [hunk 3]
```diff
@@ -36,6 +36,7 @@ class CMDArgEnum(Model):
 
     # properties as nodes
     items = ListType(ModelType(CMDArgEnumItem), min_size=1)
+    supportExtension = CMDBooleanField()
```
Comment (kairu-ms) [inline src/aaz_dev/command/model/configuration/_arg.py]: Please reference this way using `serialized_name` and `deserialize_from` https://github.com/Azure/aaz-dev-tools/blob/0276135a9dd54934581f03072c711956125282b4/src/aaz_dev/command/model/editor/_workspace.py#L29-L33

File: src/aaz_dev/command/model/configuration/_arg.py [hunk 4]
```diff
@@ -50,6 +54,7 @@ def build_enum(cls, schema_enum, ref_enum):
                         break
             item = CMDArgEnumItem.build_enum_item(schema_item, ref_enum_item)
             enum.items.append(item)
+        enum.support_extension = schema_enum.support_extension
```
Comment (kairu-ms) [inline src/aaz_dev/command/model/configuration/_arg.py]: I think this line could cause problem. Did you try reload button after set the True value in editor? After reload, the value may be false. BTW, please also verify the inherent function.

File: src/aaz_dev/command/model/configuration/_schema.py [hunk 5]
```diff
@@ -68,6 +68,10 @@ class CMDSchemaEnum(Model):
 
     # properties as nodes
     items = ListType(ModelType(CMDSchemaEnumItem), min_size=1)
+    support_extension = CMDBooleanField(
+        serialized_name="supportExtension",
+        deserialize_from="supportExtension",
+    )
```
Comment (kairu-ms) [inline src/aaz_dev/command/model/configuration/_schema.py]: It seems in current design, this property may not be needed in Schema

File: src/aaz_dev/command/controller/workspace_helper.py [hunk 6]
```diff
@@ -63,3 +66,11 @@ def _update_cls_arg(arg, **kwargs):
     def _update_array_arg(arg, **kwargs):
         if 'singularOptions' in kwargs:
             arg.singular_options = kwargs['singularOptions'] or None
+
+    @staticmethod
+    def _update_arg_enum(arg, **kwargs):
+        if 'supportEnumExtension' not in kwargs:
+            return
+        if isinstance(arg, CMDFloatArgBase) or isinstance(arg, CMDIntegerArgBase) or isinstance(arg, CMDStringArgBase):
```
Comment (kairu-ms) [inline src/aaz_dev/command/controller/workspace_helper.py]:
```suggestion
        if 'supportEnumExtension' in kwargs and isinstance(arg, (CMDFloatArgBase, CMDIntegerArgBase, CMDStringArgBase)):
```

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**